### PR TITLE
[FIX] mail: update activity view after creating multiple activities

### PR DESCRIPTION
**Pull request by sabr-odoo**
Before this commit:
When a user creates multiple scheduled activities and clicks the "Close" button the newly created activities are not updated in the activity view.

After this commit:
When a user creates multiple scheduled activities and clicks the "Close" button, the newly created activities should update and visible in the activity view.

Task-4057815

### CODE REVIEW

File: addons/mail/static/src/js/views/activity/activity_controller.js [hunk 1]
```diff
@@ -66,21 +66,30 @@ var ActivityController = BasicController.extend({
...
+                multiSelect: false,
+                context: state.context,
+                onSelected: async (resIds) => {
+                    const messaging = await owl.Component.env.services.messaging.get();
+                    const thread = messaging.models["Thread"].insert({
+                        id: resIds[0],
+                        model: this.model.modelName,
+                    });
+                    await messaging.openActivityForm({ thread });
+                    this.trigger_up("reload");
```
Comment (aab-odoo) [inline addons/mail/static/src/js/views/activity/activity_controller.js:86]:
```suggestion
```
Having the reload here as well produces 2 reloads when the activity is created via the FormViewDialog (when a row is clicked).

It seems to work fine, but don't hesitate to test a bit

**Review by aab-odoo — COMMENTED** [review]
~~The fix is correct, however~~

Oops, forgot I started to write this, before testing and seeing that the view was reloaded twice.

What I was typing was that it also requires a unit test.

**Comment by sabr-odoo** [comment]
Hello @aab-odoo ,

I have made the changes.

Thank you 🙂

**Comment by aab-odoo** [comment]
Looks good to me (didn't check the test though). I'll let @seb-odoo do the final review.

File: addons/test_mail/static/tests/activity_tests.js [hunk 2]
```diff
@@ -641,4 +641,48 @@ QUnit.test("Activity view: luxon in renderingContext", async function (assert) {
...
+    await click(document.querySelector("table tfoot tr .o_record_selector"));
+    await click(document.querySelectorAll(".o_ActivityButtonView")[1]);
+    await click(document.querySelector(".o_ActivityListView_addActivityButton"));
+    await editInput(document.querySelector('.o_field_many2one_selection input[id=activity_type_id]'), null, "test2");
+    await click(document.querySelector('.o_field_many2one_selection input[id=activity_type_id]'));
+    await click(document.querySelector('.o-autocomplete--dropdown-menu li:nth-child(1) .dropdown-item'));
+    await click(document.querySelector(".modal-footer .o_cp_buttons .o_form_buttons_edit .btn-primary"));
+    await click(document.querySelector(".modal-footer .o_form_button_cancel"));
+    assert.strictEqual(document.querySelector("tbody tr:nth-child(1) td:nth-child(7) .o_mail_activity .o_closest_deadline").textContent,
+        'Oct 15');
```
Comment (reth-odoo) [inline addons/test_mail/static/tests/activity_tests.js:685]: I think you can pass the string selector directly to `click`, to reduce noise a bit

**Review by reth-odoo — COMMENTED** [review]
~~Is this still relevant? Testing this on runbot (16) it seems adding an activity already triggers a reload~~
Didn't test it right, nevermind

Aside from that please don't do linting for a bugfix in stable. It makes blaming, forward-porting and reviewing unnecessarily more difficult.

**Comment by sabr-odoo** [comment]
Hello  @reth-odoo ,

I have made changes as you suggested, could you please take a look at it?

Thank you :slightly_smiling_face:

**Review by reth-odoo — APPROVED** [review]
Ok, I don't see a cleaner solution sadly.

Thanks for updating the test!

**Comment by kebeclibre** [comment]
@aab-odoo 
Considering the state of the discussion, I think it is ready to merge
@robodoo r+

**Pull request closed without merging**